weaviate/datadog_checks/weaviate/config_models/validators.py

```diff
@@ -0,0 +1,48 @@
+# (C) Datadog, Inc. 2023-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from urllib.parse import urlparse
+
+
+def initialize_instance(values, **kwargs):
+    if 'openmetrics_endpoint' in values:
+        validate_url(
+            values['openmetrics_endpoint'],
+            required_path='/metrics',
+            example='http://localhost:2112/metrics',
+            config='openmetrics_endpoint',
+        )
+
+    if 'weaviate_api_endpoint' in values:
+        validate_url(
+            values['weaviate_api_endpoint'],
+            required_path=None,
+            example='http://localhost:8080',
+            config='weaviate_api_endpoint',
+        )
+
+    return values
+
+
+def validate_url(url, required_path=None, example=None, config=None):
```

Comment:
nice!! I wonder if we can make this generic enough to move it as a tool into the base package at some point...